### PR TITLE
Create npm-publish.yml

### PR DESCRIPTION
**Pull request by kurayami07734**
adding github action for automating npm publish

## Need
- [ ] npm auth token

PS
I also wanted to know what all versions of node we are going to target?
(Might need to add babel)

### CODE REVIEW

**Review by oleg-nenashev — APPROVED** [review]
LGTM

**Comment by kurayami07734** [comment]
Closes #5

@oleg-nenashev we need to setup npm access token as actions secret for this action to work

**Comment by oleg-nenashev** [comment]
@kurayami07734 Deploy tokens are associated to users, so normally you should be able to add one. I will double-check what happens there. Please also add the organization scope (`@wiremock`) to the metadata: https://docs.npmjs.com/about-organization-scopes-and-packages

File: .github/workflows/npm-publish.yml [hunk 1]
```diff
@@ -0,0 +1,33 @@
...
+    steps:
+      - uses: actions/checkout@v3
+      - uses: actions/setup-node@v3
+        with:
+          node-version: 16
+          registry-url: https://registry.npmjs.org/
+      - run: npm ci
+      - run: npm publish
+        env:
+          NODE_AUTH_TOKEN: ${{secrets.npm_token}}
```
Comment (oleg-nenashev) [inline .github/workflows/npm-publish.yml:33]: The token is now configured

```suggestion
          NODE_AUTH_TOKEN: ${{secrets.NPM_PUBLISH_TOKEN}}
```

**Review by oleg-nenashev — CHANGES_REQUESTED** [review]
Requesting changes to indicate an update is needed before the merge

**Review by oleg-nenashev — APPROVED** [review]

**Pull request merged**